pre-docs/bolt_configure_puppetdb.md

```diff
@@ -0,0 +1,78 @@
+
+# Configuring PuppetDB
+
+Configure bolt to connect to PuppetDB
+
+## Why
+
+- [Generate an Inventory File](inventory_file_generating.md)
+- [Discover Facts for Targets](writing_plans.md#collect-facts-from-puppetdb)
+- [Make Queries in Plans](writing_plans.md#puppetdb_query)
+
+## PuppetDB Authorization
+
+Bolt can authenticate with PuppetDB through an ssl client certificate or a PE
+RBAC token.
+
+### Client Certificate
+
+Add the certname for the certificate you want to authenticate with to
+`/etc/puppetlabs/puppetdb/certificate-whitelist`. This certificate will have
+full access to all PuppetDB API endpoints and can read all data, push new data
+or run commands on PuppetDB. To test the certificate you can run the following
+curl command.
+
+```
+curl -X GET $SERVER_URL/pdb/query/v4 --data-urlencode 'query=nodes[certname] {}' --cert $CERT_PATH --key $KEY_PATH --cacert $CACERT_PATH
+```
+
+### PE RBAC Token
+
+If you use PE you can grant more restricted access to PuppetDB with a PE RBAC
+token. First, make sure the user you will log in as has the 'View node data from
+PuppetDB' permission. Then, login that user with `puppet-access`, this will save
+the token `~/.puppetlabs/token`. You can verify that authentication is working
+with the following curl command.
+
+```
+curl -X GET https://$SERVER_URL/pdb/query/v4 --data-urlencode 'query=nodes[certname] {}' -H "X-Authentication: `cat ~/.puppetlabs/token`" --cacert $CACERT_PATH
+```
+
+## Configuration
+
+To configure the bolt puppetdb client add a `puppetdb` section with the following values
+- `server-urls`: An array containing the PuppetDB host to connect to.
+- `cacert`: The path the ca certificate for puppetdb
+
+If you're using certificate auth also set:
+- `cert`: The path to the client certificate file to use for authentication
+- `key`: The private key for that certificate
+
+If your using a PE RBAC token set:
+- `token`: The path to the PE RBAC Token.
+
+For example to use certificate auth:
+```yaml
+puppetdb:
+  server_urls: ["https://puppet.example.com:8081"]
+  cacert: /etc/puppetlabs/puppet/ssl/certs/ca.pem
+  cert: /etc/puppetlabs/puppet/ssl/certs/my-host.example.com.pem
+  private_keys: /etc/puppetlabs/puppet/ssl/private_keys/my-host.example.com.pem
```

Comment:
s/private_keys/key/

Reply:
private_keys is correct

Reply:
Oh, weird, I get `cert and key must be specified together` if I use private_keys but no error if I use `key`. I think it's incorrect above in that case